lib/psulib_traject/call_number.rb

```diff
@@ -32,11 +34,17 @@ def reverse_shelfkey
       shelf_key.reverse
     end
 
+    def browsable?
+      return true unless lc? || dewey?
+
+      shelf_key.valid?
```

Comment:
so I tried using `shelf_key.nil?` before and could not get the result I was hoping too: eg:

```1) PsulibTraject::ShelfKey with a number that Lcsort cannot process is expected to be nil
     Failure/Error: it { is_expected.to be_nil }

       expected: nil
            got: #<PsulibTraject::ShelfKey:0x00007fb0902de338 @call_number="Fiction G758thefu 2015">
     # ./spec/lib/psulib_traject/shelf_key_spec.rb:19:in `block (3 levels) in <top (required)>'
```

The check here I need is `shelf_key.foward.nil?` so I just added a method called `valid?` to check that easier. (there I also added reserve one as well, which is redundant probably)

Reply:
I just tried changing the `valid?` method to use `nil?` rather than `is_a?`, it worked. I will make that change but let me know if you were suggesting sth else that I am missing?

Reply:
Ah ok, I got this wrong. ShelfKey won't be nil but it's forward and reverse will be. So you're solution is correct as it originally stands. However, we don't have to have `ShelfKey#valid?` at all and instead just do:

``` ruby
def browseable
  !forward_shelfkey.nil? && !reverse_shelfkey.nil?
end
```

We could also borrow an idea from Rails and add NullObject#present?

``` ruby
def present?
  false
end
```

And then have

``` ruby
def browseable
  forward_shelfkey.present? && reverse_shelfkey.present?
end
```

🤷‍♂️ 

In any case, your original solution is 💯 correct. Although, `return true unless lc? || dewey?` would be problematic because ASIS or anything else wouldn't have a shelfkey, right?

Reply:
```
def browseable
  !forward_shelfkey.nil? && !reverse_shelfkey.nil?
end
```

this is what I have in the PR currently but liked your idea for `present?`, will do that 👍🏻

Reply:
`return true unless lc? || dewey?`

I was just thinking to return true for the other types because of no shelf key 🤔